src/states/common/approved.rs

```diff
@@ -544,6 +545,57 @@ pub trait Approved: Base {
             self.send_direct_message(recipient.peer_addr(), Variant::MemberKnowledge(payload))
         }
     }
+
+    fn handle_bounce(&mut self, sender: P2pNode, sender_version: Option<u64>, msg_bytes: Bytes) {
+        if let Some((_, version)) = self.chain().find_section_by_member(sender.public_id()) {
+            if sender_version
+                .map(|sender_version| sender_version < version)
```

Comment:
If the sender of the `Bounce` is `BootstrappingPeer` or `JoiningPeer`, because it isn't a member of any section yet and so can't provide the version. In this case we always resend, because we expect them to transition to `Adult` or `Elder` shortly.